pkg/transform/client.go

```diff
@@ -0,0 +1,306 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package transform
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/bufbuild/buf/pkg/transform/internal/cache"
+	"github.com/bufbuild/buf/private/gen/proto/connect/buf/alpha/registry/v1alpha1/registryv1alpha1connect"
+	registryv1alpha1 "github.com/bufbuild/buf/private/gen/proto/go/buf/alpha/registry/v1alpha1"
+	"github.com/bufbuild/connect-go"
+)
+
+// Client ..
+type Client struct {
```

Comment:
I wonder if it's worth decomposing this a little further. Maybe we could have two different types: the `SchemaWatcher` and the `Converter`. So the client properties related to invoking the schema service RPC and caching the result could be in `SchemaWatcher`, and you then pass a `SchemaWatcher` along with from/to format options when creating a `Converter`. WDYT?